autonomy/deploy/base.py

```diff
@@ -69,33 +68,33 @@ def __init__(
         keys: Path,
         number_of_agents: Optional[int] = None,
         private_keys_password: Optional[str] = None,
+        agent_instances: Optional[List[str]] = None,
     ) -> None:
         """Initialize the Base Deployment."""
-        self.private_keys: List = []
+        self.keys: List = []
         self.private_keys_password = private_keys_password
         self.service = load_service_config(service_path)
         if number_of_agents is not None:
             self.service.number_of_agents = number_of_agents
 
+        self.agent_instances = agent_instances
         self.read_keys(keys)
-        if self.service.number_of_agents > len(self.private_keys):
-            raise ValueError("Number of agents cannot be greater than available keys.")
 
     def read_keys(self, file_path: Path) -> None:
         """Read in keys from a file on disk."""
 
         keys = json.loads(file_path.read_text(encoding=DEFAULT_ENCODING))
-
-        key_schema = (
-            KEY_SCHEMA_UNENCRYPTED_KEY
-            if self.private_keys_password is None
-            else KEY_SCHEMA_ENCRYPTED_KEY
-        )
         for key in keys:
-            for required_key in [KEY_SCHEMA_ADDRESS, key_schema]:
-                if required_key not in key.keys():
-                    raise ValueError("Key file incorrectly formatted.")
-            self.private_keys.append(key[key_schema])
+            if {KEY_SCHEMA_ADDRESS, KEY_SCHEMA_PRIVATE_KEY} != set(key.keys()):
```

Comment:
@8ball030 I don't why we store encrypted private keys as `encrypted_key` if we export them as it is and decrypt them on the agent runtime. I've update the format to simple `address` and `private_key`, Let me know if there is something that I'm missing

Reply:
for backwards compatibility, we need to support both password and non-password encrypted keys no?

Reply:
Also, we need to co-ordinate rolling this as as i will need to update all my keys and all my backups of those keys

Reply:
We still support both encrypted and non encrypted keys, just store them both as `private_key` rather than storing plain keys as `private_key` and encrypted keys as `encrypted_key`.